### PR TITLE
AC_Avoidance: get Vector3f when checking all components of relpos

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/AC_Avoidance/AP_OADatabase.cpp [hunk 1]
```diff
@@ -151,16 +151,14 @@ void AP_OADatabase::queue_push(const Vector3f &pos, uint32_t timestamp_ms, float
...
     if (!is_zero(_min_alt)) { 
-        Vector2f current_pos;
-        if (!AP::ahrs().get_relative_position_NE_home(current_pos)) {
+        Vector3f current_pos;
+        if (!AP::ahrs().get_relative_position_NED_home(current_pos)) {
             // we do not know where the vehicle is
             return;
         }
-        if (current_pos.length() < AP_OADATABASE_DISTANCE_FROM_HOME) {
+        if (current_pos.length_xy() < AP_OADATABASE_DISTANCE_FROM_HOME) {
```
Comment (hendjoshsr71) [inline libraries/AC_Avoidance/AP_OADatabase.cpp:159]:
```suggestion
        if (current_pos.xy().length() < AP_OADATABASE_DISTANCE_FROM_HOME) {
```

**Review by hendjoshsr71 — CHANGES_REQUESTED** [review]
looks good with the small fixup

File: libraries/AC_Avoidance/AP_OADatabase.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AC_Avoidance/AP_OADatabase.cpp]: Fixed

**Review by hendjoshsr71 — APPROVED** [review]

**Comment by peterbarker** [comment]
Tested in SITL by enabling the RPi simulation and checking avoidance still worked in guided / _min_alt is honoured

**Review by rishabsingh3003 — APPROVED** [review]

**Pull request merged**